design/raft.md

```diff
@@ -51,9 +51,20 @@ node's certificate, but the Raft ID is assigned arbitrarily and would change.
 
 It's important to note that a Raft ID can't be reused after a node that was
 using the ID leaves the consensus group. These Raft IDs of nodes that are no
-longer part of the cluster are saved in a list to make sure they aren't reused.
-If a node with a Raft ID on this list tries to use Raft RPCs, other nodes won't
-honor these requests.
+longer part of the cluster are saved (snapshot persisted) in a list (a blacklist,
+if you will) to make sure they aren't reused. If a node with a Raft ID on this list
+tries to use Raft RPCs, other nodes won't honor these requests. TODO why it may not
+be a good idea to not reuse raft Ids (may lead to split-brain, etc).
```

Comment:
The simplest reason is that the etcd/raft package doesn't allow ID reuse and will break badly in these situations. I think the reason it won't let us reuse IDs is that a removed node can still be running an arbitrary length of time later (for example, if it was removed during a network partition), and receiving a message from this raft ID used in two places would be ambiguous.